Code/Framework/AzFramework/AzFramework/Device/DeviceAttributeInterface.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/std/any.h>
+
+namespace AzFramework
+{
+    // DeviceAttribute provides an interface for requesting information about a
+    // single device attribute like model, GPU type, RAM amount etc.
+    class DeviceAttribute
+    {
+    public:
+        virtual ~DeviceAttribute() = default;

```

Comment:
You don't see a need for AZ_RTTI for the DeviceAttribute class?

How about casting to a derived DeviceAttribute like the DeviceAttributeDeviceModel?